### PR TITLE
TenableNessus: added io and sc dataconnectors, parser, data sample

### PR DESCRIPTION
**Pull request by averbn**
## Proposed Changes

  - added Tenable io VM connector
  - added Tenable sc VM connector

### CODE REVIEW

File: Solutions/Tenable Nessus/Data Connectors/Tenable_io_NessusVM_API_FunctionApp.json [hunk 1]
```diff
@@ -0,0 +1,129 @@
+{
+	"id":"NessusVMAPI",
+	"title":"Tenable.io Vulnerability Management Reports",
+	"publisher":"Tenable",
+	"descriptionMarkdown":"The [Tenable.io Nessus Vulnerability Management](https://www.tenable.com/products/tenable-io) Report data connector provides the capability to ingest Scan reports events into Azure Sentinel through the REST API from the Tenable.io platform (Managed in the cloud). Refer to [API documentation](https://developer.tenable.com/reference) for more information. The connector provides ability to get events which helps to examine potential security risks, analyze your team's use of collaboration, diagnose configuration problems and more",
+	"additionalRequirementBanner":"These queries and workbooks are dependent on a parser based on Kusto to work as expected. ​Follow the steps to use this Kusto functions alias **NessusVM** in queries and workbooks [Follow steps to get this Kusto functions>](https://aka.ms/)",
```
Comment (v-ampami) [inline Solutions/Tenable Nessus/Data Connectors/Tenable_io_NessusVM_API_FunctionApp.json:6]: Please replace the kusto function link https://aka.ms/  with [https://aka.ms/NessusVM](https://aka.ms/NessusVM)

File: Solutions/Tenable Nessus/Data Connectors/Tenable_sc_NessusVM_API_FunctionApp.json [hunk 2]
```diff
@@ -0,0 +1,129 @@
+{
+	"id":"NessusVMAPI",
+	"title":"Tenable.sc Vulnerability Management Reports",
+	"publisher":"Tenable",
+	"descriptionMarkdown":"The [Tenable.sc Nessus Vulnerability Management](https://www.tenable.com/products/tenable-sc) Report data connector provides the capability to ingest Scan reports events into Azure Sentinel through the REST API from the Tenable.sc vulnerability management solution (Managed on-premises). Refer to [API documentation](https://developer.tenable.com/reference) for more information. The connector provides ability to get events which helps to examine potential security risks, analyze your team's use of collaboration, diagnose configuration problems and more",
+	"additionalRequirementBanner":"These queries and workbooks are dependent on a parser based on Kusto to work as expected. ​Follow the steps to use this Kusto functions alias **NessusVM** in queries and workbooks [Follow steps to get this Kusto functions>](https://aka.ms/)",
```
Comment (v-ampami) [inline Solutions/Tenable Nessus/Data Connectors/Tenable_sc_NessusVM_API_FunctionApp.json:6]: Please replace the kusto function link https://aka.ms/ with https://aka.ms/NessusVM

**Review by v-ampami — CHANGES_REQUESTED** [review]

File: Sample Data/Custom/NessusVM_CL.json [hunk 3]
```diff
@@ -0,0 +1,92 @@
+[
```
[the remark this answers is not in the document]
Comment (averbn) [reply Sample Data/Custom/NessusVM_CL.json:1]: done

File: Solutions/Tenable Nessus/Data Connectors/Tenable_io_NessusVM_API_FunctionApp.json (hunk 1 above)
Comment (averbn) [reply Solutions/Tenable Nessus/Data Connectors/Tenable_io_NessusVM_API_FunctionApp.json:6]: done

File: Solutions/Tenable Nessus/Data Connectors/Tenable_sc_NessusVM_API_FunctionApp.json (hunk 2 above)
Comment (averbn) [reply Solutions/Tenable Nessus/Data Connectors/Tenable_sc_NessusVM_API_FunctionApp.json:6]: done

**Review by v-ampami — CHANGES_REQUESTED** [review]

File: Solutions/Tenable Nessus/Data Connectors/Tenable_sc_NessusVM_API_FunctionApp.json [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+		  {
+			 "name":"REST API Credentials/permissions",
+			 "description":"**NessusSecretKey**, **NessusAccessKey** is required for REST API. [See the documentation to learn more about API](https://<ConsoleIP>:8834/api#). Check all [requirements and follow  the instructions](https://docs.tenable.com/tenablesc/Content/GenerateAPIKey.htm) for obtaining credentials."
+		  }
+	   ]
+	},
+	"instructionSteps":[
+	   {
+		  "title":"",
+		  "description":">**NOTE:** This connector uses Azure Functions to connect to the Tenable.sc API to pull its logs into Azure Sentinel. This might result in additional data ingestion costs. Check the [Azure Functions pricing page](https://azure.microsoft.com/pricing/details/functions/) for details.\n\n This connector uses [Hybrid Connection](https://docs.microsoft.com/en-us/azure/app-service/app-service-hybrid-connections) for access to the on-premises system.\n\n In addition to there being an App Service plan SKU requirement, there is an additional cost to using Hybrid Connections. "
```
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/Tenable Nessus/Data Connectors/Tenable_sc_NessusVM_API_FunctionApp.json:75]: done

File: Solutions/Tenable Nessus/Data Connectors/Tenable_io_NessusVM_API_FunctionApp.json [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+	   },
+	   {
+		  "title":"",
+		  "description":"**2. Configure the Function App**\n\n1. In the Function App, select the Function App Name and select **Configuration**.\n2. In the **Application settings** tab, select ** New application setting**.\n3. Add each of the following application settings individually, with their respective string values (case-sensitive): \n\t\tNessusSecretKey\n\t\tNessusAccessKey\n\t\tNessusUrl\n\t\tWorkspaceID\n\t\tWorkspaceKey\n3. Once all application settings have been entered, click **Save**."
+	   }
+	]
```
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/Tenable Nessus/Data Connectors/Tenable_io_NessusVM_API_FunctionApp.json:128]: I`ll add it later

File: Solutions/Tenable Nessus/Data Connectors/Tenable_io_NessusVM_API_FunctionApp.json [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+			 "scope":"Workspace",
+			 "requiredPermissions":{
+				"write":true,
+				"read":true,
+				"delete":true
+			 }
+		  },
+		  {
+			 "provider":"Microsoft.OperationalInsights/workspaces/sharedKeys",
+			 "permissionsDisplayText":"read permissions to shared keys for the workspace are required. [See the documentation to learn more about workspace keys](https://docs.microsoft.com/azure/azure-monitor/platform/agent-windows#obtain-workspace-id-and-key)",
```
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/Tenable Nessus/Data Connectors/Tenable_io_NessusVM_API_FunctionApp.json:53]: done

File: Solutions/Tenable Nessus/Data Connectors/NessusVMAPISentinelConnector/function.json [hunk 7]
```diff
@@ -0,0 +1,11 @@
+{
+  "scriptFile": "__init__.py",
+  "bindings": [
+    {
+      "name": "mytimer",
+      "type": "timerTrigger",
+      "direction": "in",
+      "schedule": "0 0 * * * *"
```
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/Tenable Nessus/Data Connectors/NessusVMAPISentinelConnector/function.json:8]: It seems ok. (Run every 1 hour)

**Review by v-jayakal — APPROVED** [review]

**Review by v-jayakal — APPROVED** [review]

**Pull request merged**